### DISCUSSION TITLE
Updating Filters.user() in run-time

### DISCUSSION THREAD

**Discussion by AlexanderVasko** [python-telegram-bot/python-telegram-bot#2940] [Q&A] [answered]
Hi!
I have a bot with a different functionality for registered and unregistered users. Division is implemented using `Filters.user()`.
Short example:
```python
registeredListID = [11111111, 22222222, 33333333]
...
def fooRegistered(update: Update, context: CallbackContext) -> None:
    # do smth
def foo(update: Update, context: CallbackContext) -> None:
    # do smth
...
filterTxt = Filters.text & (~Filters.command)
filterRegistered = filterTxt & Filters.user(registeredListID)

dispatcher.add_handler(MessageHandler(filterRegistered, fooRegistered))
dispatcher.add_handler(MessageHandler(filterTxt, foo))
```
Users can send registration request to administrator and in case they are accepted `registeredListID` appends with a new ID but dispatcher handlers are not updating. I was trying to add new handler with new user list and same callback but it is just adding another handler and the old one is still present. Also, I was reading documentation and found out that I can edit existing handlers but it's not safe and looks like an ugly crutch.
Is there any way to update `Filters.user()` list in `MessageHandler` and other handlers in run-time without re-defining a wheel?

Thanks!

**Comment by Bibo-Joshi** [comment, marked as answer]
Yes, [`Filters.user`](https://python-telegram-bot.readthedocs.io/en/stable/telegram.ext.filters.html#telegram.ext.filters.Filters.user.add_user_ids) has some bound methods exactly for this purpose. To use them, you will have to keep a reference to that filter object, e.g.

```python
registeredListID = [11111111, 22222222, 33333333]
registered_users = Filters.user(registeredListID)
...

def fooRegistered(update: Update, context: CallbackContext) -> None:
    # do smth
def foo(update: Update, context: CallbackContext) -> None:
    registered_users.add_user(update.effective_user.id)
...
filterTxt = Filters.text & (~Filters.command)
filterRegistered = filterTxt & registered_users 

dispatcher.add_handler(MessageHandler(filterRegistered, fooRegistered))
dispatcher.add_handler(MessageHandler(filterTxt, foo))
```

**Reply by AlexanderVasko** [reply]
Oh, it seems I missed `Filters.user.add_chat_ids()`. It works well!
Thank you!
